core/src/test/java/org/ehcache/EhcacheBasicPutAllEventsTest.java

```diff
@@ -0,0 +1,216 @@
+    }
+    verify(cacheEventListener, never()).onEvent(Matchers.<CacheEvent<String, String>>any());
+    try {
+      ehcache.getRuntimeConfiguration().deregisterCacheEventListener(cacheEventListener);
+      fail();
+    } catch (UnsupportedOperationException e) {
+      //expected
+    }
+  }
+
+  @Test
+  public void testPutAllNullKey() throws Exception {
+    final Map<String, String> originalStoreContent = getEntryMap(KEY_SET_A, KEY_SET_B);
+    final FakeStore fakeStore = new FakeStore(originalStoreContent);
+    this.store = spy(fakeStore);
+
+    final Map<String, String> entries = new LinkedHashMap<String, String>();
+    for (final Map.Entry<String, String> entry : getEntryMap(KEY_SET_A).entrySet()) {
+      final String key = entry.getKey();
+      entries.put(key, entry.getValue());
+      if ("keyA2".equals(key)) {
+        entries.put(null, "nullKey");
+      }
+    }
+    final Ehcache<String, String> ehcache = this.getEhcache();
+    try {
+      ehcache.putAll(entries);
+      fail();
+    } catch (NullPointerException e) {
+      // Expected
+    }
+    verify(cacheEventListener, never()).onEvent(Matchers.<CacheEvent<String, String>>any());
+    try {
+      ehcache.getRuntimeConfiguration().deregisterCacheEventListener(cacheEventListener);
+      fail();
+    } catch (UnsupportedOperationException e) {
+      //expected
+    }
+  }
+
+  @Test
+  public void testPutAllNullValue() throws Exception {
+    final Map<String, String> originalStoreContent = getEntryMap(KEY_SET_A, KEY_SET_B);
+    final FakeStore fakeStore = new FakeStore(originalStoreContent);
+    this.store = spy(fakeStore);
+
+    final Map<String, String> entries = new LinkedHashMap<String, String>();
+    for (final Map.Entry<String, String> entry : getEntryMap(KEY_SET_A).entrySet()) {
+      final String key = entry.getKey();
+      entries.put(key, entry.getValue());
+      if ("keyA2".equals(key)) {
+        entries.put("keyA2a", null);
+      }
+    }
+    final Ehcache<String, String> ehcache = this.getEhcache();
+    try {
+      ehcache.putAll(entries);
+      fail();
+    } catch (NullPointerException e) {
+      // Expected
+    }
+    verify(cacheEventListener, never()).onEvent(Matchers.<CacheEvent<String, String>>any());
+    try {
+      ehcache.getRuntimeConfiguration().deregisterCacheEventListener(cacheEventListener);
+      fail();
+    } catch (UnsupportedOperationException e) {
+      //expected
+    }
+  }
+
+  @Test
+  public void testPutAll() throws Exception {
+    final Map<String, String> originalStoreContent = getEntryMap(KEY_SET_A, KEY_SET_B);
+    final FakeStore fakeStore = new FakeStore(originalStoreContent);
+    this.store = spy(fakeStore);
+    final Ehcache<String, String> ehcache = this.getEhcache();
+    final Map<String, String> contentUpdates = getAltEntryMap("new_", fanIn(KEY_SET_A, KEY_SET_C, KEY_SET_D));
+    ehcache.putAll(contentUpdates);
+    verify(cacheEventListener, times(18)).onEvent(argThat(isCreatedOrUpdated));
```

Comment:
maybe it would be clearer to use KEY_SET_A.size() + KEY_SET_C.size() + KEY_SET_D.size() instead of 18